docs/source/contributing.rst

```diff
@@ -0,0 +1,154 @@
+
+.. highlight:: none
+
+Getting Started
+==========================
+
+This documentation is created using 
+`Sphinx <http://www.sphinx-doc.org/en/stable>`_. Sphinx is an open-source tool 
+that makes it easy to create intelligent and beautiful documentation, written 
+by Georg Brandl and licensed under the BSD license.
+
+The documentation is maintained in the ``master`` branch of the GitHub repository.
+You can include code and its corresponding documentation updates in a single pull-request (PR).
+
+After merging a PR, GitHub Actions automates the documentation building process.
+It pushes the HTML build to the ``gh-pages`` branch, which is hosted on GitHub Pages.
+
+Edit Documentation
+===============================
+
+Sphinx uses `reStructuredText <http://docutils.sourceforge.net/rst.html>`_
+as its markup language. For more information on how to write documentation
+using Sphinx, you can refer to
+
+* `First Steps with Sphinx <http://www.sphinx-doc.org/en/stable/tutorial.html>`_
+* `reStructuredText Primer <http://www.sphinx-doc.org/en/stable/rest.html#external-links>`_
+
+1. Make sure that you have conda in your path. On NERSC machines, you can load it with: ::
+
+   $ module load python/3.7-anaconda-2019.10
+
+2. Clone the repository and checkout a branch from `master`: ::
+
+   $ cd <myDir>
+   $ git clone https://github.com/<your-github-username>/zstash.git
+   $ cd zstash
+   $ git checkout -b <branch-name> master
+
+3. Create and activate the conda development environment ::
+
+   $ cd <myDir>/zstash
+   $ conda env create -f conda/dev.yml
+   $ conda activate zstash_env_dev
+
+4. To modify the documentation, simply edit the files under ``/docs/source``.
+
+5. To see the changes you made to the documentation, rebuild the web pages ::
+
+   $ cd <myDir>/zstash/docs
+   $ make html
+ 
+6. View them locally in a web browser at ``file:///<myDir>/zstash/index.html``.
+ - Sometimes the browser caches Sphinx docs, so you might need to delete your cache to view changes.
+
+7. Once you are satisfied with your modifications, commit and push them back to the repository: ::
+
+    $ cd <myDir>/zstash
+    $ git add .
+    $ git commit
+    $ git push origin <branch-name>
+
+8. <`OPTIONAL`> If you want to generate and view versioned docs: ::
+
+    $ # After commiting to your branch
+    $ cd <myDir>/zstash/docs
+    $ sphinx-multiversion source _build/html
+    $ # Check the `_build/html` folder for all generated versioned docs
+    $ # Open `_build/html/<your-branch>/index.html` to view in browser
+
+   .. figure:: _static/docs-version-selector.png
+      :alt: Docs version selector
+
+      Docs version selector dropdown in the bottom left-hand corner
+
+9. Create a pull-request from ``your-fork/zstash/branch-name`` to ``E3SM-Project/zstash/master``.
+
+Once this pull-request is merged and GitHub Actions finishes building the docs, changes will be available on the
+`zstash documentation page <https://e3sm-project.github.io/zstash/>`_.
+
+
```

Comment:
Everything above this line should be reviewed. I successfully followed the steps locally and on Cori.